### ISSUE TITLE
Some details

### ISSUE THREAD

**Issue by zbb1111** [linyq2117/CLIP-ES#17]
Sorry to bother you again, could you please tell me about the innovative points you proposed in which files and which part of the code?
Like:
1. Sharpness based prompt?
2. Synonym fusion?
3. CAA?

**Comment by linyq2117** [comment]
1. The prompt we used is `a clean origami {}.`, which can be found at [here](https://github.com/linyq2117/CLIP-ES/blob/d0d0c07b73b44c66a13dda5cdbe9f751fddb6b05/generate_cams_voc12.py#L236).
2. The synonyms of class names are shown in `clip_text.py`.
3. The CAA is conducted at [these lines](https://github.com/linyq2117/CLIP-ES/blob/d0d0c07b73b44c66a13dda5cdbe9f751fddb6b05/generate_cams_voc12.py#L168C17-L188C49).

**Comment by zbb1111** [comment]
Thank you very much and wish you all the best!

**Comment by lk0429** [comment]
Excuse me for disturbing, could you please tell me which part of the code implements the Confidence-guided Loss (CGL)?

**Comment by linyq2117** [comment]
> Excuse me for disturbing, could you please tell me which part of the code implements the Confidence-guided Loss (CGL)?

Thanks for your interest in our work!

We do not explicitly adopt confidence-guided loss. Instead, we set the segmentation mask to 255 if the max confidence of this pixel is less than 0.95 for convenience. The label 255 will be ignored by nn.CrossEntropyLoss in deeplab-pytorch. This process is implemented in [eval_cam_with_crf.py](https://github.com/linyq2117/CLIP-ES/blob/d0d0c07b73b44c66a13dda5cdbe9f751fddb6b05/eval_cam_with_crf.py#L123).

**Comment by linyq2117** [comment]
Closed for inactivity.
